pkg/kube/kubehttp.go

```diff
@@ -37,12 +37,12 @@ const WeGONamespace = "wego-system"
 const WeGOCRDName = "apps.wego.weave.works"
 const FluxNamespace = "flux-system"
 
-func NewKubeHTTPClient() (Kube, error) {
+func NewKubeHTTPClient() (Kube, client.Client, error) {
```

Comment:
The `client.Client` utilized by the `AuthService`

Reply:
>  I'd leave it as part of the structure and remove it as an individual argument.

Sorry I originally misread. The `kube.Kube` implementation does not always have the `client.Client` as a dependency, only the `KubeHTTP` struct has it internally.

The other issue is that `client.Client` makes the entire `kube.Kube` interface unnecessary, so there is probably a bigger change that needs to be made around this interface.

Reply:
In fact, given the relatively small number of calls we make to `client.Client`, we could just have the `KubeHTTP` struct implement an interface that extends `kube.Kube` and not expose the raw client at all. That would let us mock those calls if we want to as well so we wouldn't always have to spin up a k8s server. Also, it would make our code more approachable to a potential open source contributor. They would know exactly what we do with Kubernetes. If we pass a raw client around, they have to assume we might do literally anything supported by Kubernetes.

Reply:
I would say that `client.Client` is just as easily mock-able (with some code generation), and gives us the added option of using the `envtest` package for more integration-y, non-mocked tests. T[he interfaces are also broken up by reader/writer if we want to get more specific](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Client).

Reply:
Closing for now. I assume we will discuss further before changing that interface.